manager-rs/dauth-service/src/rpc/handler.rs

```diff
@@ -74,46 +81,655 @@ impl LocalAuthentication for DauthHandler {
+        &self,
+        request: tonic::Request<FloodVectorReq>,
+    ) -> Result<tonic::Response<FloodVectorResp>, tonic::Status> {
+        tracing::info!("Request: {:?}", request);
+
+        let message = request
+            .into_inner()
+            .message
+            .ok_or_else(|| tonic::Status::new(tonic::Code::NotFound, "No message received"))?;
+
+        let verify_result =
+            signing::verify_message(self.context.clone(), &message).or_else(|e| {
+                Err(tonic::Status::new(
+                    tonic::Code::Unauthenticated,
+                    format!("Failed to verify message: {}", e),
+                ))
+            })?;
+
+        match DauthHandler::flood_vector_hlp(self.context.clone(), verify_result).await {
+            Ok(result) => Ok(result),
+            Err(e) => Err(tonic::Status::new(
+                tonic::Code::Aborted,
+                format!("Error while handling request: {}", e),
+            )),
+        }
+    }
+}
+
+/// Implementation of all helper functions to reuse/condense error logic
+impl DauthHandler {
+    /* General helpers */
+    pub fn handle_delegated_vector(
+        context: Arc<DauthContext>,
+        dvector: DelegatedAuthVector5G,
+        user_id: &str,
+    ) -> Result<AuthVectorRes, DauthError> {
+        let verify_result = signing::verify_message(
+            context.clone(),
+            &dvector.message.ok_or(DauthError::InvalidMessageError(
+                "Missing content".to_string(),
+            ))?,
+        )?;
+
+        if let SignPayloadType::DelegatedAuthVector5G(payload) = verify_result {
+            Ok(AuthVectorRes::from_av5_g(
+                user_id,
+                payload.v.ok_or(DauthError::InvalidMessageError(
+                    "Missing content".to_string(),
+                ))?,
+            )?)
+        } else {
+            Err(DauthError::InvalidMessageError(format!(
+                "Incorrect message type: {:?}",
+                verify_result
+            )))
+        }
+    }
+
+    pub fn handle_key_share(
+        context: Arc<DauthContext>,
+        dshare: DelegatedConfirmationShare,
+    ) -> Result<(auth_vector::types::HresStar, auth_vector::types::Kseaf), DauthError> {
+        let verify_result = signing::verify_message(
+            context.clone(),
+            &dshare.message.ok_or(DauthError::InvalidMessageError(
+                "Missing content".to_string(),
+            ))?,
+        )?;
+
+        if let SignPayloadType::DelegatedConfirmationShare(payload) = verify_result {
+            Ok((
+                payload.xres_star_hash[..].try_into()?,
+                payload.confirmation_share[..].try_into()?,
+            ))
+        } else {
+            Err(DauthError::InvalidMessageError(format!(
+                "Incorrect message type: {:?}",
+                verify_result
+            )))
+        }
+    }
+
+    /* Specific helpers */
+    pub async fn get_home_auth_vector_hlp(
```

Comment:
This helper functions probably should not be public? Or are they called from outside DauthHander?

Reply:
Good point! I'll remove those